include/wx/private/smartrowheightcache.h

```diff
@@ -0,0 +1,142 @@
+///////////////////////////////////////////////////////////////////////////////
+// Name:        wx/private/smartrowheightcache.h
+// Purpose:     Smart height cache of rows in a dataview
+// Author:      Jens Goepfert
+// Created:     2018-03-06
+// Copyright:   (c) 2018 Vadim Zeitlin <vadim@wxwidgets.org>
+// Licence:     wxWindows licence
+///////////////////////////////////////////////////////////////////////////////
+
+#ifndef _WX_PRIVATE_SMARTROWHEIGHTCACHE_H_
+#define _WX_PRIVATE_SMARTROWHEIGHTCACHE_H_
+
+#include "wx/hashmap.h"
+#include "wx/vector.h"
+
+// struct describing a range of rows which contains rows <from> .. <to-1>
+struct RowRange
+{
+    unsigned int from;
+    unsigned int to;
+};
+
+/**
+@class RowRanges
+A helper class that manages a set of RowRange objects.
+It stores the indices that are members of a group in a memory
+efficiant way.
+*/
+class WXDLLIMPEXP_ADV RowRanges
+{
+public:
+    /**
+    Adds a row index to this group by adding it to an existing RowRange
+    or by creating a new one.
+    */
+    void Add(const unsigned int row);
```

Comment:
Please don't use top level `const`, it's useless and inconsistent with the rest of wxWidgets code.

Reply:
i updated this